SCClassLibrary/Common/Math/Integer.sc

```diff
@@ -171,6 +171,14 @@ Integer : SimpleNumber {
 		*/
 	}
 
+	factorial {
+		if(this > 12) {
+			"Insufficient integer precision for this number (factorial(%)). "
+			"Try to use float instead.".format(this).warn
+		};
+		^super.factorial.asInteger
```

Comment:
I had originally thought it best to return the correct `Float` rather than the "incorrect" `Integer` or limiting the resolution by throwing an error for the sake of preserving the type.
For example, returning a float looks like:
```
35.factorial;
-> 1.0333147966386e+40
```
while casting to an `Integer` will return
```
-> -2147483648
```
(along with a warning)

Is it preferable to return an incorrect `Integer` rather than a correct `Float`?

Reply:
I think both of those solutions would be better than warning and returning something incorrect.

Reply:
No, I'd prefer it to be the `Float`, but we need to take care not to break existing code.

Reply:
… actually, it might be better even to return nothing and throw an error.

Reply:
I am fine with either option. At the moment I have a slight preference toward throwing an error on failure; returning a float on some inputs and an int on others seems like a good way to confuse users and in the case of an error it's easy to trace.